test/exe/Main.hs

```diff
@@ -1532,7 +1533,12 @@ run s = withTempDir $ \dir -> do
   -- HIE calls getXgdDirectory which assumes that HOME is set.
   -- Only sets HOME if it wasn't already set.
   setEnv "HOME" "/homeless-shelter" False
-  runSessionWithConfig conf cmd fullCaps { _window = Just $ WindowClientCapabilities $ Just True } dir s
+  let lspTestCaps =
```

Comment:
`lsp-test` has a generic `sendNotification` function that we should be able to use to send `didChangeWatchedFiles` events.

Reply:
Thanks, I didn't realize that. 

After enabling the WatchedFiles capability in the test client, an existing test for adding a missing module no longer passed, and with the use of `sendNotification` to send a `WatchedFile` file creation event, now does. 

From my perspective the new feature is now sufficiently tested, but I suppose if ghcide really wanted to support clients without WatchedFiles capability, the entire test suite should be run twice with and without the capability, or at least have tests for both cases.